### PR TITLE
OpenSource Projects Pages

### PR DESCRIPTION
**Pull request by gastonsoto**
Adding main and individual pages.

### CODE REVIEW

File: app/views/home/opensource.html.erb [hunk 1]
```diff
@@ -1,5 +1,91 @@
...
+    <div class="row">
+    <% @opensource_projects.each do |project| %>
+      <div class="col-md-4">
+        <div class="card mb-4 shadow-sm hvr-grow">
+          <div class="card-header"
+               style="background:<%= project[:image][:background] %>">
+              <a href="/opensource/<%= project[:slug] %>" class="d-flex h-100">
+                <img class="d-block w-50 m-auto align-self-center"
+                    src="<%= project[:image][:src] %>" 
+                    alt="<%= project[:name] %>">
```
Comment (martinlevesque) [inline app/views/home/opensource.html.erb]: the name should probably always be printed.

File: app/views/home/opensource.html.erb [hunk 2]
```diff
@@ -1,5 +1,91 @@
...
+  <div class="container">
+    <div class="row">
+    <% @opensource_projects.each do |project| %>
+      <div class="col-md-4">
+        <div class="card mb-4 shadow-sm hvr-grow">
+          <div class="card-header"
+               style="background:<%= project[:image][:background] %>">
+              <a href="/opensource/<%= project[:slug] %>" class="d-flex h-100">
+                <img class="d-block w-50 m-auto align-self-center"
+                    src="<%= project[:image][:src] %>"
```
Comment (martinlevesque) [inline app/views/home/opensource.html.erb]: the image should be optional, don't think everyone will have a logo. And so when there is no image the name should probably be bigger (?)

File: config/routes.rb [hunk 3]
```diff
@@ -16,6 +16,7 @@
   get '/terms', to: 'home#terms'
   get '/privacy', to: 'home#privacy'
   get '/opensource', to: 'home#opensource'
```
Comment (martinlevesque) [inline config/routes.rb]: add a dash -> open-source

File: app/views/home/opensource.html.erb [hunk 4]
```diff
@@ -1,5 +1,91 @@
...
   <div class="container cointaier-shrinked">
-    <h1 class="display-4 text-left">OpenSource</h1>
+    <h1 class="display-4 text-left text-success">OpenSource Projects</h1>
+    <h2 class="text-muted text-left">We offer free hosting to selected open source projects.</h2>
   </div>
-</section>
\ No newline at end of file
+</section>
+<section class="pt-8 pt-md-11 bg-light p-0 p-md-5">
+  <div class="container carousel-opensource d-none d-md-block">
```
Comment (martinlevesque) [inline app/views/home/opensource.html.erb]: it's possible to have no featured project (@featured_projects = [])

File: app/views/home/opensource.html.erb [hunk 5]
```diff
@@ -1,5 +1,91 @@
...
+         href="#carouselIndicators"
+         role="button"
+         data-slide="next">
+        <i class="fa fa-2x fa-chevron-right text-secondary" aria-hidden="true"></i>
+        <span class="sr-only">Next</span>
+      </a>
+    </div>
+  </div>
+</section>
+<section class="opensource-list pt-8 pt-md-11 bg-secondary p-3 p-md-5">
```
Comment (martinlevesque) [inline app/views/home/opensource.html.erb]: would be nice to have a condition, if no open source project, print a big "coming soon" -> there won't be any on day 1

maybe render a specific partial page when there is no project

Comment (gastonsoto) [reply app/views/home/opensource.html.erb]: Added.

File: app/views/home/opensource.html.erb (hunk 4 above)
Comment (gastonsoto) [reply app/views/home/opensource.html.erb]: Done.

File: config/routes.rb (hunk 3 above)
Comment (gastonsoto) [reply config/routes.rb]: Done.

File: app/views/home/opensource.html.erb (hunk 2 above)
Comment (gastonsoto) [reply app/views/home/opensource.html.erb]: Done. Added a conditional for the logo. Displaying the name if the logo src is blank.

**Comment by gastonsoto** [comment]
Updated.

**Pull request merged**